src/stream/src/executor_v2/hash_agg.rs

```diff
@@ -176,41 +167,41 @@ impl<K: HashKey, S: StateStore> AggHashAggExecutor<K, S> {
         Ok(result)
     }
 
-    fn is_dirty(&self) -> bool {
-        self.state_map
-            .values()
-            .any(|state| state.as_ref().unwrap().is_dirty())
-    }
-}
-
-#[async_trait]
-impl<K: HashKey, S: StateStore> AggExecutor for AggHashAggExecutor<K, S> {
-    async fn apply_chunk(&mut self, chunk: StreamChunk, epoch: u64) -> StreamExecutorResult<()> {
+    #[allow(clippy::too_many_arguments)]
+    async fn apply_chunk(
+        key_indices: &[usize],
+        agg_calls: &[AggCall],
+        input_pk_indices: &[usize],
+        input_schema: &Schema,
+        schema: &Schema,
+        state_map: &mut EvictableHashMap<K, Option<Box<AggState<S>>>>,
+        keyspace: &Keyspace<S>,
+        chunk: StreamChunk,
+        epoch: u64,
```

Comment:
There're too many arguments, and I don't think it's a good idea...
If you just want to avoid the ownership problem in `execute_inner`, we may use an `Option` for `self.input` and take it first.

Reply:
I have a good idea, I’ll try to fix it later.

Reply:
> I have a good idea, I’ll try to fix it later.

I've applied the idea of `Option` for `self.input` in current PR. Maybe future fixes can be opened in another PR. 👍